### PR TITLE
Remove unused pragma: no cover

### PR DESCRIPTION
**Pull request by maxnordlund**
`sed -i "" 's/  # pragma: no cover$//' tests/cover/*`

This is the follow up PR from #481

### CODE REVIEW

**Comment by maxnordlund** [comment]
No problem, `sed` is your friend :)

**Comment by DRMacIver** [comment]
I've just seen the failing build and I'm really sorry. :-(

I think what's going on here is that that branch was currently only covered semi by accident because the code for testing it had a comment in the middle. Sigh.

**Comment by maxnordlund** [comment]
Damn it. OK what do we do then?

**Comment by DRMacIver** [comment]
> Damn it. OK what do we do then?

Least-effort solution is to ignore the problem and back out the change. :-)

If you're feeling motivated, second least effort solution is to add an assert False to the currently find which line in test\_lambda\_formatting.py triggers that assert if you add a comment after it. Then a) Duplicate that test with the comment and b) Figure out what it was supposed to be testing and see if you can make it test that properly.

> I seem to have a knack for finding weird bugs, oh the joy

It's more that he code you're currently touching is very weird and bad. :-( It has that annoying property of being code that serves a useful function that there's no really good way to implement. 

I've tried to replace it at least once with something more reliable, but it didn't go very well and I had to revert that change.

**Comment by maxnordlund** [comment]
I don't quite follow, which line is supposed to have an `assert False`? Is there a HTML report with the coverage, or a way to get one to see what's missing?

**Comment by DRMacIver** [comment]
There's no HTML but there's a report printed at the end. The relevant line is here: https://travis-ci.org/HypothesisWorks/hypothesis-python/jobs/214336605#L1534, so this line is uncovered: https://github.com/HypothesisWorks/hypothesis-python/blob/master/src/hypothesis/internal/reflection.py#L244

**Comment by maxnordlund** [comment]
Ah, I saw that but didn't see the line numbers.

> If you're feeling motivated, second least effort solution is to add an assert False to the currently find which line in test_lambda_formatting.py triggers that assert if you add a comment after it. 

I still can digest that sentence though. Would you mind explain it a bit more?

**Comment by DRMacIver** [comment]
> Would you mind explain it a bit more?

Sure. So the idea is that the easiest way to detect whether a test covers a line is to replace the line with an assert False, so turning hitting it into an error.

We know that adding a comment of "#pragma: no cover" on at least one of those tests will cause that line to be triggered. So if you just add those comments back in and run the tests then you'll find which one does that.

At that point you have a test that reliably triggers the line (the version with the comment). You can then adapt that to a test that triggers the line using a different coment once you've removed the pragmas again.

You can possibly also see what that test was *supposed* to be triggering and make sure there's a test that still does, but I'm not necessarily sure that will be relevant..

**Comment by maxnordlund** [comment]
OK, now I get it. I'll give it a go right away then.

**Comment by maxnordlund** [comment]
Actually I don't get any failing tests, just `ERROR: InvocationError: '~/.cache/hypothesis-build-runtimes/.tox/coverage/bin/coverage report -m --fail-under=100 --show-missing'` but I think that's unrelated. It did print out the report as well.

**Comment by Zac-HD** [comment]
[This is the failing coverage check](https://travis-ci.org/HypothesisWorks/hypothesis-python/jobs/214336605#L1534), indicating that line 244 is not executed by any test (and iirc that it could be called from line 233).  If it's obvious to you why (it isn't to me), feel free to fix it; either way you'll need to rebase on master to fix the conflict.

**Comment by maxnordlund** [comment]
OK, so I finally got a counter example to trigger that line. Turns out it was quite hard, see the commit message for more.

Now it gives 100% test coverage locally, so let's just hope Travis agrees.

**Comment by maxnordlund** [comment]
Ping @Zac-HD and @DRMacIver

File: tests/cover/test_lambda_formatting.py [hunk 1]
```diff
@@ -61,35 +68,35 @@ def test_does_not_error_on_dynamically_defined_functions():
...
 def test_is_not_confused_by_tuples():
-    p = (lambda x: x > 1, 2)[0]  # pragma: no cover
+    p = (lambda x: x > 1, 2)[0]
 
     assert get_pretty_function_description(p) == 'lambda x: x > 1'
 
 
 def test_strips_comments_from_the_end():
-    t = lambda x: 1  # pragma: no cover
+    t = lambda x: 1
```
Comment (Zac-HD) [inline tests/cover/test_lambda_formatting.py]: I think you need a different comment here, judging by the name of the test.

Comment (maxnordlund) [inline tests/cover/test_lambda_formatting.py]: Indeed, I've fixed it now.

**Pull request merged**